lib/sidekiq/form526_job_status_tracker/backup_submission.rb

```diff
@@ -0,0 +1,38 @@
+# frozen_string_literal: true
+
+module Sidekiq
+  module Form526JobStatusTracker
+    module BackupSubmission
+      def send_backup_submission_if_enabled(form526_submission_id:, job_class:, job_id:, error_class:, error_message:)
+        submission_obj ||= Form526Submission.find(form526_submission_id)
+        additional_birls_to_try = submission_obj.birls_ids_that_havent_been_tried_yet
+        backup_job_jid = nil
+        flipper_sym = :form526_backup_submission_temp_killswitch
+        # Entry-point for backup 526 CMP submission
+        #
+        # Required criteria to send a backup 526 submission from here:
+        # Enabled in settings and flipper
+        # Is an overall submission and NOT an upload attempt
+        # Does not have a valid claim ID (through RRD process or otherwise) (protect against dup submissions)
+        # Does not have a backup submission ID (protect against dup submissions)
+        # Does not have additional birls it is going to try and submit with
+        send_backup_submission = Settings.form526_backup.enabled && Flipper.enabled?(flipper_sym) &&
```

Comment:
Oh I see what you are saying, bail on pulling the 526 if either of the other disqualifying checks are false. Yea that makes sense I will do that.

Reply:
Fixed formatting and failing test in bfe7065a96b80d4c6ca962e06813c5ea643b1848

Reply:
So guard clauses, disqualifying returns, etc are a commonly used anti pattern in ruby because they obfuscate behavior.  ¯\_(ツ)_/¯ Ruby methods "should" have a single return.  They are not the worst thing, but in a perfect world we should avoid using them.  However, since a major refactor of the existing logic is out of scope, not worth getting hung up on  here.

Reply:
> Ruby methods "should" have a single return

Yea, this is usually true. However in places where that is not the case, I have usually seen it OK as long as the different returns are explicit and not implied returns. At least that is what I have always tried to stick to. 


There are example of this in the Ruby style guide that allow/support/prefer multiple returns and such.

![image](https://github.com/department-of-veterans-affairs/vets-api/assets/37049625/95608be9-753d-4f03-a9ea-cf614df3b782)